File: QTD.SPA/src/app/_DtoModels/ClientSettingsLabelReplacement/ClientSettings_LabelReplacement_UpdateOptions.ts
export class ClientSettings_LabelReplacement_UpdateOptions {

  labelReplacements: LabelReplacementOptions[];

  UpdateLabelReplacement = (defaultLabel: string, labelReplacement: string) => {

    if (!this.labelReplacements) this.labelReplacements = [];

    let labelReplacementOption = this.labelReplacements.filter(r => r.defaultLabel === defaultLabel)[0];

    if (!labelReplacementOption) {
      labelReplacementOption = new LabelReplacementOptions(defaultLabel, labelReplacement);
      this.labelReplacements.push(labelReplacementOption);
    } else {
      labelReplacementOption.SetLabelReplacementvalue(labelReplacement);
    }
  }
}

export class LabelReplacementOptions {
  defaultLabel: string;
  labelReplacement: string;

  constructor(defaultLabel, labelReplacement) {
    this.labelReplacement = labelReplacement;
    this.defaultLabel = defaultLabel;
  }

  SetLabelReplacementvalue = (labelReplacement) => {
    this.labelReplacement = labelReplacement;
  }
}

File: QTD2.Infrastructure/Model/ClientSettings/ClientSettings_LabelReplacementUpdateOptions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.ClientSettings
{
    public class ClientSettings_LabelReplacementUpdateOptions
    {
        public List<LabelReplacementOptions> LabelReplacements { get; set; }
    }

    public class LabelReplacementOptions
    {
        public string DefaultLabel { get; set; }
        public string LabelReplacement { get; set; }
    }
}
